### ISSUE TITLE
reload and check logical in arbiter

### ISSUE THREAD

**Issue by andyxning** [naparuba/shinken#1545]
i found that no matter what reload(without -v) or check(-v) is specified with shinken-arbiter, we just do the [is-correct](https://github.com/naparuba/shinken/blob/master/shinken/daemons/arbiterdaemon.py#L446) check, this may be useless and may make the reload take more time. 

I think we can add a more option with shinken-arbiter **-i**, i.e., ignore check. By default we can use the is_correct function to check if the configuration is right. But if we can make sure that the configuration is already checked and everything is fine, we can **explicitly** add the **-i** option to bypass the  check logic which has already been done with **-v** options.

**Comment by naparuba** [comment]
Object check do not take lot of time as far as I remeber. Please give use
hints about how much time we can save here by skipping this phase on a
normal launch after a manual check.

**Comment by geektophe** [comment]
I agree with @naparuba, the check phase is negligible in regard to the parse phase, which is anyway mandatory. In addition, even if the I could, personaly I would not take the risk to send a configuration which has not been verified before.

What could be done to optimize restart time could be to reload the configuration from the arbiter process itself on the HUP signal, and if it's correct, replace the existing one. This could avoid having to restart the process, and perform a double check.

**Comment by andyxning** [comment]
i agree with @Christophe Simon, that we should take consider to make HUP
signal handler available on arbiter and in that case without restart the
arbiter process we can have arbiter to process external commands with no
lose and still do other works.



++++++
Ning Xie


2015-03-27 2:18 GMT+08:00 Christophe Simon <notifications@github.com>:

**Comment by geektophe** [comment]
I'll have a look at that.

**Comment by andyxning** [comment]
Can we just make this **Reload configuration without stop Arbiter daemon** as a main feature in Shinken 2.6. I think it is really useful and  essential

**Comment by naparuba** [comment]
Yes, if the new arbiter is start with the -r parameter, it will kill the
old one *after* loading and parsing all elements, so the huge part. That
can make the old arbiter alive for longer period and less "no arbiter" time
:)

**Comment by andyxning** [comment]
That's very great and i am looking for that feature.

++++++
Ning Xie


2015-04-23 14:44 GMT+08:00 Gabès Jean <notifications@github.com>:
